### PR TITLE
fix(key-manager): remove floating point math from mnemonic code

### PR DESCRIPTION
**Pull request by sdbondi**
Description
---
- removes floating-point math from `mnemonic::from_bytes`
- masks "last" byte before conversion to u8 (clippy)

Motivation and Context
---
This fixes clippy cast warnings and avoids the overhead of working with floating-points 

How Has This Been Tested?
---
This module is tested well, these tests pass.

### CODE REVIEW

File: base_layer/key_manager/src/mnemonic.rs [hunk 1]
```diff
@@ -185,20 +185,22 @@ pub fn from_bytes(bytes: &[u8], language: MnemonicLanguage) -> Result<Vec<String
 
     // Pad with zeros if length not divisible by 11
     let group_bit_count = 11;
-    let padded_size = ((bits.len() as f32 / group_bit_count as f32).ceil() * group_bit_count as f32) as usize;
+    let mut padded_size = bits.len() / group_bit_count;
```
Comment (Cifko) [inline base_layer/key_manager/src/mnemonic.rs:188]: If you want, you can get rid of the `mut` by doing just `let padded_size = bits.len() + group_bit_count - 1 - (bits.len() - 1) % group_bit_count;`

File: base_layer/key_manager/src/mnemonic.rs [hunk 2]
```diff
@@ -216,36 +218,37 @@ pub fn to_bytes(mnemonic_seq: &[String]) -> Result<Vec<u8>, MnemonicError> {
...
 /// We now have enough bits to get some output, we take 8 bits and produce output byte. We do this as long as we have at
 /// least 8 bits in the 'rest'.
 /// Sample of couple first steps:
 /// 1) the first output 'a' is last 8 bits from input 'A', we have leftover 3 bits from 'A'
 /// 2) We add 5 bits from 'B' to generate 'b', the leftover is 6 bits from 'B'
 /// 3) We add 2 bits from 'C to generate 'c', now we have 8 bits needed to generate 'd' and we have 1 bit leftover.
 pub fn to_bytes_with_language(mnemonic_seq: &[String], language: &MnemonicLanguage) -> Result<Vec<u8>, MnemonicError> {
-    let mut bytes: Vec<u8> = Vec::new();
-    let mut rest = 0;
+    const MASK: u64 = (1u64 << 8) - 1;
```
Comment (Cifko) [inline base_layer/key_manager/src/mnemonic.rs:232]: you can use `u8::MAX.into()`, but it's matter of preference.

**Review by Cifko — APPROVED** [review]
Tested with `cargo test`

File: base_layer/key_manager/src/mnemonic.rs (hunk 1 above)
Comment (SWvheerden) [reply base_layer/key_manager/src/mnemonic.rs:188]: https://doc.rust-lang.org/std/primitive.usize.html#method.div_ceil
Does exactly what we want, but it's currently nightly only.

**Review by stringhandler — APPROVED** [review]

**Review by SWvheerden — APPROVED** [review]
5 minutes, 
Tested on new and old wallet

File: base_layer/key_manager/src/mnemonic.rs (hunk 2 above)
Comment (sdbondi) [reply base_layer/key_manager/src/mnemonic.rs:232]: Good point, like that better but oh well :)

**Pull request merged**